Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -1210,6 +1210,7 @@ public boolean cast(SpellAbility originalAbility, Game game, boolean noMana, App
                 castEvent.setZone(fromZone);
                 game.fireEvent(castEvent);
                 if (spell.activate(game, noMana)) {
+                    game.applyEffects();
```

Comment:
It's a magic that can ruin game cycle at all. See comments from #8402 for game cycle details and applyEffects/processAction.

`game.applyEffects()` already called in `spell.activate`'s code

![shot_221128_025834](https://user-images.githubusercontent.com/8344157/204164502-e943e273-70a9-477f-b272-22797e075293.png)

Need more research and possible fix. I don't recommends to add `applyEffects` for any fixes without critical reason.

Reply:
I did end up changing this to `game.getState().processAction(game);`. Please let me know if this is acceptable.

`game.applyEffects()` is already called in `spell.activate`'s code, but this is before costs are paid. Please take a look at [my comment](https://github.com/magefree/mage/issues/9669#issuecomment-1328336013) in the linked issue (https://github.com/magefree/mage/issues/9669) for details.